app/controllers/cart_controller.rb

```diff
@@ -24,5 +24,16 @@ def remove_item
     redirect_to '/cart'
   end
 
+  def add_more_item
+    item = Item.find(params[:item_id])
+
+      if item.inventory_has_reached_limit?(cart, item)
+        flash[:error] = "You've reached the end of this merchant's inventory"
+        redirect_to '/cart'
+      else
+        cart.add_item(item.id.to_s)
+        redirect_to '/cart'
+      end
+  end
```

Comment:
Nice method. I like the model method you have to check if the inventory is full

Reply:
Thank you!